### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by ctokoli**
### These changes were done on this branch 

- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```
Comment (tresorsawasawa) [inline book.rb]: - [ ] As required for the project to correctly respect  the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. kindly ensure that: `Getters` and `Setters` for `@rentals` are present, `@rentals` should also be present in your constructor method as instance variable, and you should also have a method for adding rentals where the book itself is sent as a parameter to create the new rental achieving the 'has-many' association. It should be similar to this:
```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .
  def initialize(title, author)
    .
    .
    @rentals = []
  end
  
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: person_class.rb [hunk 2]
```diff
@@ -4,14 +4,15 @@
...
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   private
```
Comment (tresorsawasawa) [inline person_class.rb]: - [ ] As required for the project the a person to have many rentals, kindly ensure that: `Getters` and `Setters` for `@rentals` are present, and you should also have a method for adding rentals where the person itself should is sent as a parameter to create the new rental achieving the 'has-many' association. It should be similar to this:
```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
    @rentals = []
  end

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @ctokoli 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- Most of associations are correctly set up :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (ctokoli) [reply book.rb]: Hello @tresorsawasawa, Thanks for the review but I think the requirements stated what should be done clearly and if am not doing it right then the project requirement is not right. This is the requirement and i can’t see the request to create a method `Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).`.  Anyway, I will go ahead with your suggestions.  Thanks

Comment (tresorsawasawa) [reply book.rb]: Hi @ctokoli 

As the `@rentals` depends on your relationship that why I mentioned that, you can even check all the requirements
Screenshot from 2023-09-13 14-01-07

**Review by DammyShittu — APPROVED** [review]
Hi @ctokoli,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**